### ISSUE TITLE
Is this a bug?  Ubuntu14.04 i686 configured terminal program '' not found or not executable

### ISSUE THREAD

**Issue by VOID001** [Lazy-Newb-Pack/LNP-Linux-legacy#12]
When I download the game ,install all the dependencies and run the **./startlnp** command from terminal (with and without tmux) then click the Button **Play Dwarf Fortress**
the following error occured:

```
xdg-terminal: configured terminal program '' not found or not executable
```

And when I open the stderr file  it says:

```
Exception in Tkinter callback
Traceback (most recent call last):
  File "/home/michael/Desktop/df/python_lnp/build/lnp/out00-PYZ.pyz/Tkinter", line 1486, in __call__
  File "/home/michael/Desktop/df/python_lnp/build/lnp/out00-PYZ.pyz/core.launcher", line 50, in run_df
Exception: Failed to run Dwarf Fortress.
```

I am sure I input the right sudo password and installed all the dependencies in README

Below is my Linux kernel Version and my Ubuntu is **14.04**

Linux void133-ThinkPad-T430 3.13.0-32-generic #57-Ubuntu SMP Tue Jul 15 03:51:12 UTC 2014 i686 i686 i686 GNU/Linux

My home dir is /home/void-admin/     

Could you please help me figure out what the problem is? Thanks in advance

ps : here is the snapshot when I run ./startlnp and then Click **Play Dwarf Fortress**
![image](https://cloud.githubusercontent.com/assets/6134883/5714832/2dd1c3f8-9b09-11e4-8469-c6c963fa3ce1.png)
![image](https://cloud.githubusercontent.com/assets/6134883/5714838/43644740-9b09-11e4-84be-06b02d7fdea0.png)

**Comment by BeauBouchard** [comment]
Thank you for taking the time to write an error report. I work during the week, but i will be looking into this error. 

If you don't mind me asking a few questions, are you able to launch Dwarf Fortress on its own?
 (located in /home/void-admin/pack/df_linux/ )

**Comment by VOID001** [comment]
Thank you for replying  ^_^
**I can launch it** and I'm now playing by lauching from the df_linux directory
and there seems to be some problems when I launch the utilities from LNPack.
The gentoopatch and DwarfTherapist cannot launch and 
I tried to launch gentoopatch directly here is the problem
![image](https://cloud.githubusercontent.com/assets/6134883/5754344/8df269ca-9cd2-11e4-8727-f71d414bba35.png)

and if I use chmod then and then launch it from the shell directly 
![image](https://cloud.githubusercontent.com/assets/6134883/5754362/c9a8f5d8-9cd2-11e4-98fc-a0283589eac8.png)
And then I try again to launch it from LNPack and it still dosen't work

and I  launch DwarfTherapist directly with the same way above, the error report looks like this

```
bash: ./DwarfTherapist: cannot execute binary file: Exec format error
```

**Comment by jezeniel** [comment]
I get the same `stderr.txt` output above.. but mine only says
`/bin/sh: symbol lookup error: /bin/sh: undefined symbol: rl_signal_event_hook`. Any thoughts about this?

**Comment by ghost** [comment]
I have a similar issue in Linux Mint Debian Edition. I use `terminator` as my terminal and have removed `gnome-terminal`. I receive:

`xdg-terminal: configured terminal program 'gnome-terminal' not found or not executable`

This happens because PyLNP's xdg-terminal script checks for the desktop environment's default, and since Debian is GNOME, it tries to launch `gnome-terminal` instead of the user's actual default terminal.

**Comment by miffedmap** [comment]
@VOID001 , Everything in the pack which is compiled seems to be compiled for 64bits right now and will not run on a 32bit distro such as you have installed. df itself is 32 bit which is why you can run it directly. 

@cptmashek - if you have:

```
$ gconftool-2  --get /desktop/gnome/applications/terminal/exec
gnome-terminal
```

Try to set the default terminal to your preferred app:

```
$ gconftool-2 --type string --set /desktop/gnome/applications/terminal/exec terminator
```

@jezenielzapanta , that might stem from an incompatibility with a recent-ish readline version,
it may come from several sources, would help of you can narrow down the program that emits that
error. It might be an old version of python was used as the basis for the compiled gui.

You might try [python-lnp](https://bitbucket.org/Pidgeot/python-lnp) directly instead of the compiled version, but that might involve some fiddling with things.

**Comment by VOID001** [comment]
Oh, I see,Thanks a lot
@miffedmap

**Comment by miffedmap** [comment]
@VOID001, I just noticed the [releases](https://github.com/Lazy-Newb-Pack/Lazy-Newb-Pack-Linux/releases/) page does contain packs for i686 as well as x86_64, you just need to grab the right zipball from there and you should be ok after all.

**Comment by miffedmap** [comment]
I think this can be closed. The gentoopatch issue is no longer relevent, the `rl_` issue should be fixed soon through #1 and using amd64 pack on an i686 box by mistake isn't actually a bug.

for everyone with a bad terminal getting launched: it's probably an issue with the system configuration but in any case PyLNP includes a menu item for overriding this manually (found under the File menu).
